plugins/inventory/nb_inventory.py

```diff
@@ -461,23 +476,52 @@ def get_resource_list(self, api_url):
 
     @property
     def group_extractors(self):
-        return {
-            "sites": self.extract_site,
-            "tenants": self.extract_tenant,
-            "racks": self.extract_rack,
-            "tags": self.extract_tags,
-            "disk": self.extract_disk,
-            "memory": self.extract_memory,
-            "vcpus": self.extract_vcpus,
-            "device_roles": self.extract_device_role,
-            "platforms": self.extract_platform,
-            "device_types": self.extract_device_type,
-            "services": self.extract_services,
-            "config_context": self.extract_config_context,
-            "manufacturers": self.extract_manufacturer,
-            "interfaces": self.extract_interfaces,
-            "custom_fields": self.extract_custom_fields,
-        }
+
+        if self.plurals:
```

Comment:
Do we need this logic here? It appears each of these functions call the `pluralize` function to then return a list or just the value.

Reply:
The keys in the dictionary are different for each option - determines what the `host_vars` will be named, and what `group_by` options are valid.

Reply:
Ok that makes sense.